pydap_extras/handlers/csv.py

```diff
@@ -0,0 +1,242 @@
+    The order of the variables can be changed:
+        >>> for line in seq['temperature', 'site', 'index']:
+        ...     print(line)
+        (15.2, 'Diamond_St', 10.0)
+        (13.1, 'Blacktail_Loop', 11.0)
+        (13.3, 'Platinum_St', 12.0)
+        (12.1, 'Kodiak_Trail', 13.0)
+    We can iterate over children:
+        >>> for line in seq['temperature']:
+        ...     print(line)
+        15.2
+        13.1
+        13.3
+        12.1
+    We can filter the data:
+        >>> for line in seq[ seq.index > 10 ]:
+        ...     print(line)
+        (11.0, 13.1, 'Blacktail_Loop')
+        (12.0, 13.3, 'Platinum_St')
+        (13.0, 12.1, 'Kodiak_Trail')
+        >>> for line in seq[ seq.index > 10 ]['site']:
+        ...     print(line)
+        Blacktail_Loop
+        Platinum_St
+        Kodiak_Trail
+        >>> for line in seq['site', 'temperature'][ seq.index > 10 ]:
+        ...     print(line)
+        ('Blacktail_Loop', 13.1)
+        ('Platinum_St', 13.3)
+        ('Kodiak_Trail', 12.1)
+    Or slice it:
+        >>> for line in seq[::2]:
+        ...     print(line)
+        (10.0, 15.2, 'Diamond_St')
+        (12.0, 13.3, 'Platinum_St')
+        >>> for line in seq[ seq.index > 10 ][::2]['site']:
+        ...     print(line)
+        Blacktail_Loop
+        Kodiak_Trail
+        >>> for line in seq[ seq.index > 10 ]['site'][::2]:
+        ...     print(line)
+        Blacktail_Loop
+        Kodiak_Trail
+    Finally, delete the data file:
+        >>> temp_file.remove()
+
+    """
+    def __init__(self, filepath, template, ifilter=None, imap=None,
+                 islice=None, level=0):
+        self.filepath = filepath
+        self.template = template
+        self.level = level
+
+        self.ifilter = ifilter or []
+        self.imap = imap or []
+        self.islice = islice or []
+
+    @property
+    def stream(self):
+        """Generator that yield lines of the file."""
+        try:
+            fp = open(self.filepath, 'Ur')
+        except Exception as exc:
+            message = f'Unable to open file {self.filepath}: {exc}'
+            raise OpenFileError(message)
+
+        reader = csv.reader(fp, quoting=csv.QUOTE_NONNUMERIC)
+        next(reader)  # consume var names
+        for row in reader:
+            yield row
+        fp.close()
+
+    def __copy__(self):
+        """Return a lightweight copy."""
+        return self.__class__(self.filepath, copy.copy(self.template),
+                              self.ifilter[:], self.imap[:], self.islice[:],
+                              self.level)
+
+
+def _test():
+    import doctest
+    doctest.testmod()
+
+
+if __name__ == "__main__":
+    import sys
+    from werkzeug.serving import run_simple
+
+    _test()
+
+    application = CSVHandler(sys.argv[1])
+    from pydap.wsgi.ssf import ServerSideFunctions
+    application = ServerSideFunctions(application)
+    run_simple('localhost', 8001, application, use_reloader=True)
```

Comment:
Have you considered moving this to a separate module? Usually running a simple version of the app will be done in a `wsgi.py` or `app.py` module.